CMakeLists.txt

```diff
@@ -22,3 +35,9 @@ target_link_libraries(pong glm)
 # and I would like to keep /WX as well. Since I don't have an easy way to ignore the warnings
 # for the system libraries, just use /W4. Seems like /Wall has some weird extra warnings anyways.
 target_compile_options(pong PRIVATE /W4 /WX)
+
+target_compile_definitions(pong PRIVATE
+    $<$<CONFIG:Debug>:
+        DEBUG=1
```

Comment:
One thing I still don't understand entirely is Debug vs Release modes in cmake. Is it something specfically predefined by cmake? Is there anything else it does maybe under the hood? I got this code from somewhere else, seems like if it is debug mode it sets the DEBUG flag.

Reply:
Yep you are right in your interpretation. If you want to learn more, this is called a "generator expression" and is similar to a ternary expression in C++. For example:

```
    bool isDebug = DEBUG_DEFINED ? 1 : 0;
```

Is kind of what is going on here:

If the first argument into the outer expression is 1, you replace it with the text, otherwise you replace it with zero, so here, in order you have:

```
$<$<CONFIG:Debug>:Debug=1>>

Evaluate:
$<CONFIG:Debug>

If it is true:
$<1:Debug=1>

is the first argument 1? Yes, so this statement becomes:

Debug=1;


```

https://jeremimucha.com/2021/03/cmake-generator-expressions/